src/Symfony/Component/Security/Http/Firewall/UsernamePasswordJsonAuthenticationListener.php

```diff
@@ -73,6 +73,9 @@ public function __construct(TokenStorageInterface $tokenStorage, AuthenticationM
     public function handle(GetResponseEvent $event)
     {
         $request = $event->getRequest();
+        if ('json' !== $request->getRequestFormat() || 'json' !== $request->getContentType()) {
```

Comment:
I would use `false !== strpos($format, 'json')` instead of the strict test to allow to execute the listener for any JSON-based formats including JSON-LD and HAL.